Add time evolution driver with jax-based ODE solver (#1012)
netket/netket: CHANGELOG.md

* Chunking of `MCState.expect` and `MCState.expect_and_grad` computations is now supported, which allows to bound the memory cost in exchange of a minor increase in computation time. [#1006](https://github.com/netket/netket/pull/1006) (and discussions in [#918](https://github.com/netket/netket/pull/918) and [#830](https://github.com/netket/netket/pull/830))
* {ref}`nk.operator.LocalOperator` now accepts sparse matrices as input operators [#919](https://github.com/netket/netket/pull/919)
* A new variational state that performs exact summation over the whole Hilbert space has been added. It can be constructed with {ref}`nk.vqs.ExactState` and supports the same Jax neural networks as {ref}`nk.vqs.MCState`. [#953](https://github.com/netket/netket/pull/953)

> * A new time-evolution driver that uses the Time-dependent-variational-principle has been added to the experimental submodule as {ref}`nk.experimental.TDVP`. It works with time-independent and time-dependent hamiltonians and liouvillians. [#1012](https://github.com/netket/netket/pull/1012) 
> * A set of Runge-Kutta, jax-compatible integrators has been added to the experimental module to use together with the new TDVP driver. [#1012](https://github.com/netket/netket/pull/1012) 

Author: ```suggestion
* [Experimental] A new time-evolution driver  {ref}`nk.experimental.TDVP` using the time-dependent variational principle (TDVP) has been added. It works with time-independent and time-dependent Hamiltonians and Liouvillians. [#1012](https://github.com/netket/netket/pull/1012) 
* [Experimental] A set of JAX-compatible Runge-Kutta ODE integrators has been added for use together with the new TDVP driver. [#1012](https://github.com/netket/netket/pull/1012) 
```
